src/components/WidgetWrapper.tsx

```diff
@@ -0,0 +1,88 @@
+import { createContext, PropsWithChildren, useEffect, useMemo, useRef, useState } from 'react'
+import styled from 'styled-components/macro'
+
+const HORIZONTAL_PADDING = 8
+
+const StyledWidgetWrapper = styled.div<{ width?: number | string }>`
+  -moz-osx-font-smoothing: grayscale;
+  -webkit-font-smoothing: antialiased;
+  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+  background-color: ${({ theme }) => theme.container};
+  border-radius: ${({ theme }) => theme.borderRadius}em;
+  box-sizing: border-box;
+  color: ${({ theme }) => theme.primary};
+  display: flex;
+  flex-direction: column;
+  font-size: 16px;
+  font-smooth: always;
+  font-variant: none;
+  min-height: 360px;
+  min-width: 300px;
+  padding: ${HORIZONTAL_PADDING}px;
+  position: relative;
+  user-select: none;
+  width: ${({ width }) => width && (isNaN(Number(width)) ? width : `${width}px`)};
+
+  * {
+    box-sizing: border-box;
+    font-family: ${({ theme }) => (typeof theme.fontFamily === 'string' ? theme.fontFamily : theme.fontFamily.font)};
+
+    @supports (font-variation-settings: normal) {
+      font-family: ${({ theme }) => (typeof theme.fontFamily === 'string' ? undefined : theme.fontFamily.variable)};
+    }
+  }
+`
+
+type WidgetWidthContextType = {
+  widgetWidth: number
+}
+
+export const WidgetWidthContext = createContext<WidgetWidthContextType>({
+  widgetWidth: 0,
```

Comment:
If this _only_ contains widgetWidth, it does not need to be an object, it can just be a number:

    const WidgetWidthContext = createContext<number>(0)